### ISSUE TITLE
Language Server textDocumentSync are limited

### ISSUE THREAD

**Issue by tm1000** [vimeo/psalm#6008]
Is there a reason the language server only sync's change events? Looking at https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/LanguageServer/Server/TextDocument.php I can see the code exists to allow openClose and save

However only `change` is declared as an ability (https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/LanguageServer/LanguageServer.php#L231). This seems to have been the case from the first day this was added?

Since only change is supported static analysis is not run on an open file until a user changes something in that document (or alternatively hover is triggered).

This wouldn't be an issue because maybe you'd only want analysis to run if the file is modified but that breaks because it runs anyways whenever any text is hovered over so why not just enable openClose

Can we enable the other methods? 

```
        "textDocumentSync": {
            "openClose": null,
            "change": 1,
            "willSave": null,
            "willSaveWaitUntil": null,
            "save": null
        }
```

I will submit a PR for this in a moment

**Comment by weirdan** [comment]
Wouldn't it make sense to just call `$this->server->doAnalysis()` there?

**Comment by tm1000** [comment]
@weirdan confused. Call that method where?

**Comment by weirdan** [comment]
In `didOpen()` and `didSave()` handlers.

**Comment by tm1000** [comment]
Ah. I can try that. I haven’t tested those functions yet because they aren’t called by the client because the server explicitly says it doesn’t suppose those functions. Even though it seems to. Let me do some work and get back to you with probably a pr

**Comment by tm1000** [comment]
So the code that already exists works as expected if I simply just add

```
                $textDocumentSyncOptions->openClose = TextDocumentSyncKind::FULL;
                $textDocumentSyncOptions->save = TextDocumentSyncKind::FULL;
```

The files are then analyzed through `queueFileAnalysis`

**Comment by tm1000** [comment]
@weirdan so these methods were originally written by @muglug so I assume them to be "pure" code. Once enabled I encountered no issues.
